62.CAD/vertex_shader.hlsl

```diff
@@ -336,17 +346,79 @@ PSInput main(uint vertexID : SV_VertexID)
+        }
+
+        const double2 ndcAabbExtents = double2(
+            length(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMax.x, curveBox.aabbMin.y) - curveBox.aabbMin)),
+            length(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMin.x, curveBox.aabbMax.y) - curveBox.aabbMin))
+        );
+        // Max corner stores the quad's UVs:
+        // (0,1)|--|(1,1)
+        //      |  |
+        // (0,0)|--|(1,0)
+        double2 maxCorner = double2(bool2(vertexIdx & 0x1u, vertexIdx >> 1));
+        
+        // Anti-alising factor + 1px due to aliasing with the bbox (conservatively rasterizing the bbox, otherwise
+        // sometimes it falls outside the pixel center and creates a hole in major axis)
+        // The AA factor is doubled, so it's dilated in both directions (left/top and right/bottom sides)
+        const double2 dilatedAabbExtents = ndcAabbExtents + 2.0 * ((globals.antiAliasingFactor + 1.0) / double2(globals.resolution));
+        // Dilate the UVs
+        maxCorner = ((((maxCorner - 0.5) * 2.0 * dilatedAabbExtents) / ndcAabbExtents) + 1.0) * 0.5;
+        const double2 coord = transformPointNdc(clipProjectionData.projectionToNDC, lerp(curveBox.aabbMin, curveBox.aabbMax, maxCorner));
+        outV.position = float4((float2) coord, 0.f, 1.f);
+
+        const uint major = (uint)SelectedMajorAxis;
+        const uint minor = 1-major;
+
+        nbl::hlsl::shapes::Quadratic<double> curveMin = nbl::hlsl::shapes::Quadratic<double>::construct(
+            curveBox.curveMin[0], curveBox.curveMin[1], curveBox.curveMin[2]);
+        nbl::hlsl::shapes::Quadratic<double> curveMax = nbl::hlsl::shapes::Quadratic<double>::construct(
+            curveBox.curveMax[0], curveBox.curveMax[1], curveBox.curveMax[2]);
+
+        outV.setMinorBBoxUv(maxCorner[minor]);
+        outV.setMajorBBoxUv(maxCorner[major]);
+
+        outV.setCurveMinMinor(nbl::hlsl::equations::Quadratic<float>::construct(
+            (float)curveMin.A[minor], 
+            (float)curveMin.B[minor], 
+            (float)curveMin.C[minor]));
+        outV.setCurveMinMajor(nbl::hlsl::equations::Quadratic<float>::construct(
+            (float)curveMin.A[major], 
+            (float)curveMin.B[major], 
+            (float)curveMin.C[major]));
+
+        outV.setCurveMaxMinor(nbl::hlsl::equations::Quadratic<float>::construct(
+            (float)curveMax.A[minor], 
+            (float)curveMax.B[minor], 
+            (float)curveMax.C[minor]));
+        outV.setCurveMaxMajor(nbl::hlsl::equations::Quadratic<float>::construct(
+            (float)curveMax.A[major], 
+            (float)curveMax.B[major], 
+            (float)curveMax.C[major]));
+
+        nbl::hlsl::equations::Quadratic<float> curveMinRootFinding = nbl::hlsl::equations::Quadratic<float>::construct(
+            (float)curveMin.A[major], 
```

Comment:
remove to float casts here, you'll be getting uint values and unpacking them into floats. and the curveMin/Max you construct above should be with float